modules/core/comment/farm_comment.module

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * @file
+ * Contains farm_comment.module.
+ */
+
+use Drupal\comment\Plugin\Field\FieldType\CommentItemInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\entity\BundleFieldDefinition;
+
+/**
+ * Implements hook_entity_base_field_info().
+ */
+function farm_comment_entity_base_field_info(EntityTypeInterface $entity_type) {
+  $fields = [];
+
+  // Add comment base field to assets and logs.
+  if (in_array($entity_type->id(), ['asset', 'log'])) {
+    $fields['comment'] = farm_comment_base_field_definition($entity_type->id());
+  }
+
+  return $fields;
+}
+
+/**
+ * Helper function for generating a standard comment base field definition.
+ *
+ * @param string $entity_type
+ *   The entity type.
+ *
+ * @return \Drupal\Core\Field\BaseFieldDefinition
+ *   Returns a comment base field definition.
+ */
+function farm_comment_base_field_definition(string $entity_type) {
+
+  // Create a new comment field definition.
+  // We use BundleFieldDefinition instead of BaseFieldDefinition to force Drupal
+  // to create a separate database table for this field. Otherwise, if it is
+  // added to the base table then the comment field default value is always 0
+  // (CommentItemInterface::HIDDEN) instead of 2 (CommentItemInterface::OPEN),
+  // because the Drupal\comment\Plugin\Field\FieldType\CommentItem::schema()
+  // default is 0.
+  $field = BundleFieldDefinition::create('comment');
+
+  // Set the field label.
+  $field->setLabel(t('Comments'));
+
+  // We assume that the comment type is "[entity-type]_comment".
+  $field->setSetting('comment_type', $entity_type . '_comment');
```

Comment:
I don't think the `_comment` suffix is necessary for the comment type bundle ID, it will always be in the context of the comment entity type. Could we simplify this so the comment type is only the entity type?

As an example, this is the API path right now: `/api/comment/log_comment`

Reply:
https://github.com/mstenta/farmOS/commit/0354fe2d3702d716da7a3fc5fc5a4674b96b8c23